server/src/com/thoughtworks/go/server/web/BackupFilter.java

```diff
@@ -0,0 +1,173 @@
+import com.thoughtworks.go.server.service.BackupService;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+import org.springframework.web.util.HtmlUtils;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @understands redirecting all requests to a service unavailable page when the server is being backed up.
+ */
+public class BackupFilter implements Filter {
+
+    public static final String JSON = "json";
+    public static final String XML = "xml";
+
+    @Autowired
+    private BackupService backupService;
+    private final static org.apache.log4j.Logger LOGGER = org.apache.log4j.Logger.getLogger(BackupFilter.class);    // For the Test
+    public static final Pattern PATTERN = Pattern.compile("^((.*/api/.*)|(.*[^/]+\\.(xml|json)(\\?.*)?))$");
+
+    // For the Test
+    protected BackupFilter(BackupService backupService) {
+        this.backupService = backupService;
+    }
+
+    public BackupFilter() {
+
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        if (backupService == null) {
+            SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        }
+
+        String url = ((HttpServletRequest) request).getRequestURI();
+        ((HttpServletResponse) response).setHeader("Cache-Control", "private, max-age=0, no-cache");
+        ((HttpServletResponse) response).setDateHeader("Expires", 0);
+
+        if (isBackupFinishJsonUrl(url)) {
+            generateResponseForIsBackupFinishedAPI(request, response);
+            return;
+        }
+        if (backupService.isBackingUp()) {
+            if (isAPIUrl(url) && !isMessagesJson(url)) {
+                generateAPIResponse(request, response);
+            } else {
+                generateHTMLResponse(response);
+            }
+        } else {
+            chain.doFilter(request, response);
+        }
+
+    }
+
+    private void generateHTMLResponse(ServletResponse response) {
+        String path = "backup_in_progress.html";
+        InputStream resourceAsStream = getClass().getClassLoader().getResourceAsStream(path);
+        response.setContentType("text/html");
+        StringWriter writer = new StringWriter();
+        try {
+            IOUtils.copy(resourceAsStream, writer);
+            String content = writer.toString();
+            content = replaceStringLiterals(content);
+            response.getWriter().print(content);
```

Comment:
Prefer `FileUtils.readToEnd(InputStream)`? I think you may want to close the inputstream inside `FileUtils` as well.